[next-devel] Add platform-specific console configuration metadata

bgilbert opened the pull request:
coreos-assembler `create_disk.sh` and `gf-set-platform` will read this.  `create_disk.sh` will also save a JSON version for the relevant architecture to `/boot/coreos/platforms.json` for coreos-installer.

Continuation of #1181.

dustymabe commented:
We could go ahead and get this in `rawhide` now if we wanted some advance testing.

bgilbert commented:
Rebased.  Moved tests to https://github.com/coreos/fedora-coreos-config/pull/1974.  Re-added graphical consoles to AWS, Azure, and GCP on x86_64, since those platforms allow taking screenshots of the graphical console to diagnose the VM.

dustymabe commented:
We've got the major cloud platforms covered here. Any other ones we think we need to add?

bgilbert commented:
I've rechecked all of the FCOS platforms against any documentation provided by the platform provider.  The `ibmcloud` serial console apparently slipped through last time, so I've now added that.

dustymabe commented:
any sane defaults we could use for openstack?

bgilbert commented:
We're using them.  OpenStack can support both graphical and serial console, but graphical console appears to be the one that's emphasized by the UI.
